### PR TITLE
Revamping Delete, Install, Uninstall, Change, Add and Update commands along with new "k8s" scope in API Controller

### PR DESCRIPTION
**Pull request by wasuradananjith**
## Purpose
Fixes: https://github.com/wso2/product-apim-tooling/issues/293
Fixes: https://github.com/wso2/product-apim-tooling/issues/340
Fixes: https://github.com/wso2/product-apim-tooling/issues/309

## Goals
Revamp the commands in the format below and deprecate the existing commands.
(A special scope for Kubernetes will be added as "k8s" in the commands related to K8s Operator)
**Format:- _apictl \<scope> \<action> \<resource> \<parameters>_**

## Approach
- Copied the code related to the existing commands (specially related to K8s Operator) inside that directory.
- Modified the code in the **cmd** directory to match with the format _apictl \<scope> \<action> \<resource> \<parameters>_.
- Moved the common code segments to the **impl** and refer those from **deprecated** package and **cmd** package both.
- Refactored the code

## User stories
- Please find the revamped commands addressed by this PR.
<table>
  <tr>
    <th>Old Command</th>
    <th>Revamped Command</th>
  </tr>
  <tr>
    <td>apictl delete api [flags]</td>
    <td>apictl delete api [flags] <br>
           apictl k8s delete api [flags]</td>
  </tr>
  <tr>
    <td>apictl install api-operator [flags]</td>
    <td>apictl k8s install api-operator [flags]</td>
  </tr>
  <tr>
    <td>apictl install wso2am-operator [flags]</td>
    <td>apictl k8s install wso2am-operator [flags]</td>
  </tr>
  <tr>
    <td>apictl uninstall api-operator [flags]</td>
    <td>apictl k8s uninstall api-operator [flags]</td>
  </tr>
  <tr>
    <td>apictl uninstall wso2am-operator [flags]</td>
    <td>apictl k8s uninstall wso2am-operator [flags]</td>
  </tr>
  <tr>
    <td>apictl change registry [flags]</td>
    <td>apictl k8s change registry [flags]</td>
  </tr>
  <tr>
    <td>apictl add api [flags]</td>
    <td>apictl k8s add api [flags]</td>
  </tr>
  <tr>
    <td>apictl update api [flags]</td>
    <td>apictl k8s update api [flags]</td>
  </tr>
  <tr>
    <td>apictl add-env [flags]</td>
    <td>apictl add env [flags]</td>
  </tr>
</table>

- Deprecated the flag "mode" in "apictl set [flags]"

## Test environment
- Ubuntu 20.04.1 LTS
- go version go1.14.4 linux/amd64

## Related PRs
https://github.com/wso2/product-apim-tooling/pull/515

### CODE REVIEW

File: import-export-cli/impl/changeAPIStatus.go [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+// Get the ID of an API if available
+// @param accessToken : Token to call the Publisher Rest API
+// @param name : Name of the API
+// @param version : Version of the API
+// @param provider : Provider of the API
+// @param environment : Environment where the API resides
+// @return apiId, error
+func getAPIIdForStateChange(accessToken, name, version, provider, environment string, credential credentials.Credential) (string, error) {
+	// Unified Search endpoint from the config file to search APIs
+	unifiedSearchEndpoint := utils.GetUnifiedSearchEndpointOfEnv(environment, utils.MainConfigFilePath)
```
Comment (malinthaprasan) [inline import-export-cli/impl/changeAPIStatus.go:77]: Do we need a special method for this? I remember there were several number of methods to get API. Its better to move those similar methods to a single package and make it reusable. I guess this method does nothing more than getting the ID of the API by doing a search from name and version?

Comment (malinthaprasan) [reply import-export-cli/impl/changeAPIStatus.go:77]: There is a method called getAPIID in impl.importAPI.go.. Can we make it public and reuse it instead?

Also, its better to keep single responsibility across the files then we can easily find methods we need and can make it easily reusable. We don't usually expect a public method GetAPIID inside changeAPIStatus.go or importAPI.go. A better place would be having it in a file similar to apis.go or api.go.

Comment (malinthaprasan) [reply import-export-cli/impl/changeAPIStatus.go:77]: There is another getAPIId method inside deleteAPI.go

Comment (wasuradananjith) [reply import-export-cli/impl/changeAPIStatus.go]: Fixed via https://github.com/wso2/product-apim-tooling/pull/518/commits/d268de6195b81448eb0d7a780e0610316ab2b37b, https://github.com/wso2/product-apim-tooling/pull/518/commits/944b4e515b1372bf691e8574e4c58484b55aeece and https://github.com/wso2/product-apim-tooling/pull/518/commits/f5cb477f64ff9f04f524065a00cd3a68be80d97a

Comment (malinthaprasan) [reply import-export-cli/impl/changeAPIStatus.go]: Great!

**Review by malinthaprasan — APPROVED** [review]

**Pull request merged**